src/Mod/PartDesign/Gui/CommandPrimitive.cpp

```diff
@@ -64,80 +79,36 @@ void CmdPrimtiveCompAdditive::activated(int iMsg)
     if (!PartDesignGui::assureModernWorkflow(doc))
         return;
 
-    PartDesign::Body *pcActiveBody = PartDesignGui::getBody(true);
-
-    if (!pcActiveBody)
-        return;
-
-    Gui::ActionGroup* pcAction = qobject_cast<Gui::ActionGroup*>(_pcAction);
-    pcAction->setIcon(pcAction->actions().at(iMsg)->icon());
+    // We need either an active Body, or for there to be no Body objects
+    // (in which case, just make one) to make a new additive shape.
 
-    std::string FeatName;
-    if(iMsg == 0) {
+    PartDesign::Body *pcActiveBody = PartDesignGui::getBody( /* messageIfNot = */ false );
 
-        FeatName = getUniqueObjectName("Box");
-
-        Gui::Command::openCommand("Make additive box");
-        Gui::Command::doCommand(Gui::Command::Doc,"App.ActiveDocument.addObject(\'PartDesign::AdditiveBox\',\'%s\')",
-            FeatName.c_str());
-    }
-    else if(iMsg == 1) {
-
-        FeatName = getUniqueObjectName("Cylinder");
-
-        Gui::Command::openCommand("Make additive cylinder");
-        Gui::Command::doCommand(Gui::Command::Doc,"App.ActiveDocument.addObject(\'PartDesign::AdditiveCylinder\',\'%s\')",
-            FeatName.c_str());
-    }
-    else if(iMsg == 2) {
-
-        FeatName = getUniqueObjectName("Sphere");
-
-        Gui::Command::openCommand("Make additive sphere");
-        Gui::Command::doCommand(Gui::Command::Doc,"App.ActiveDocument.addObject(\'PartDesign::AdditiveSphere\',\'%s\')",
-            FeatName.c_str());
-    }
-    else if(iMsg == 3) {
-
-        FeatName = getUniqueObjectName("Cone");
-
-        Gui::Command::openCommand("Make additive cone");
-        Gui::Command::doCommand(Gui::Command::Doc,"App.ActiveDocument.addObject(\'PartDesign::AdditiveCone\',\'%s\')",
-            FeatName.c_str());
-    }
-    else if(iMsg == 4) {
-
-        FeatName = getUniqueObjectName("Ellipsoid");
-
-        Gui::Command::openCommand("Make additive ellipsoid");
-        Gui::Command::doCommand(Gui::Command::Doc,"App.ActiveDocument.addObject(\'PartDesign::AdditiveEllipsoid\',\'%s\')",
-            FeatName.c_str());
-    }
-    else if(iMsg == 5) {
-
-        FeatName = getUniqueObjectName("Torus");
-
-        Gui::Command::openCommand("Make additive torus");
-        Gui::Command::doCommand(Gui::Command::Doc,"App.ActiveDocument.addObject(\'PartDesign::AdditiveTorus\',\'%s\')",
-            FeatName.c_str());
+    auto shouldMakeBody( false );
```

Comment:
Fair comment.  There are essentially two reasons I do that:

1. Folks that know a lot more about C++ than me say it's a good idea. https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/
2. I spend a lot of time working in other-than-C++, and find that it's helpful to keep each language style distinct.  Otherwise I tend to write everything like it's C, which doesn't help anybody in other-than-C projects.